servers/src/main/java/tachyon/worker/file/FileDataManager.java

```diff
@@ -116,7 +175,10 @@ public synchronized void persistFile(long fileId, List<Long> blockIds) throws IO
     outputStream.flush();
     outputChannel.close();
     outputStream.close();
-    mPersistedFiles.add(fileId);
+    synchronized (this) {
```

Comment:
The idea is to have a lock for both `mPersistingInProgressFiles` and `mPersistedFiles`. Another way than using `this` is to add another lock field.